huaweicloud/services/codearts/resource_huaweicloud_codearts_deploy_host.go

```diff
@@ -0,0 +1,455 @@
+	updatePath := client.Endpoint + "v2/host-groups/{group_id}/hosts/{host_id}"
+	updatePath = strings.ReplaceAll(updatePath, "{group_id}", d.Get("group_id").(string))
+	updatePath = strings.ReplaceAll(updatePath, "{host_id}", d.Id())
+	updateOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+		MoreHeaders: map[string]string{
+			"Content-Type": "application/json;charset=utf-8",
+		},
+		JSONBody: buildUpdateDeployHostBodyParams(d),
+	}
+
+	updateResp, err := client.Request("PUT", updatePath, &updateOpt)
+	if err != nil {
+		return fmt.Errorf("error updating CodeArts deploy host: %s", err)
+	}
+
+	updateRespBody, err := utils.FlattenResponse(updateResp)
+	if err != nil {
+		return err
+	}
+
+	if err := checkResponseError(updateRespBody, hostNotFound); err != nil {
+		return fmt.Errorf("error updating CodeArts deploy host: %s", err)
+	}
+	return nil
+}
+
+func buildUpdateDeployHostBodyParams(d *schema.ResourceData) map[string]interface{} {
+	bodyParams := map[string]interface{}{
+		"host_name":       d.Get("name"),
+		"ip":              d.Get("ip_address"),
+		"port":            d.Get("port"),
+		"as_proxy":        d.Get("as_proxy"),
+		"install_icagent": d.Get("install_icagent"),
+		"sync":            d.Get("sync"),
+		"authorization":   buildAuthorizationBodyParam(d),
+	}
+
+	if v, ok := d.GetOk("proxy_host_id"); ok {
+		bodyParams["proxy_host_id"] = v
+	}
+
+	return bodyParams
+}
+
+func resourceDeployHostDelete(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	var (
+		cfg     = meta.(*config.Config)
+		region  = cfg.GetRegion(d)
+		httpUrl = "v2/host-groups/{group_id}/hosts/{host_id}"
+		product = "codearts_deploy"
+	)
+	client, err := cfg.NewServiceClient(product, region)
+	if err != nil {
+		return diag.Errorf("error creating CodeArts deploy client: %s", err)
+	}
+
+	deletePath := client.Endpoint + httpUrl
+	deletePath = strings.ReplaceAll(deletePath, "{group_id}", d.Get("group_id").(string))
+	deletePath = strings.ReplaceAll(deletePath, "{host_id}", d.Id())
+	deleteOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+		MoreHeaders: map[string]string{
+			"Content-Type": "application/json;charset=utf-8",
+		},
+	}
+
+	deleteResp, err := client.Request("DELETE", deletePath, &deleteOpt)
+	if err != nil {
+		return diag.Errorf("error deleting CodeArts deploy host: %s", err)
+	}
+
+	deleteRespBody, err := utils.FlattenResponse(deleteResp)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	if err := checkResponseError(deleteRespBody, hostNotFound); err != nil {
+		return common.CheckDeletedDiag(d, err, "error deleting CodeArts deploy host")
+	}
+
+	return nil
+}
+
+func resourceDeployHostImportState(_ context.Context, d *schema.ResourceData,
+	_ interface{}) ([]*schema.ResourceData, error) {
+	parts := strings.Split(d.Id(), "/")
+	if len(parts) != 2 {
+		return nil, fmt.Errorf("invalid format specified for import ID,"+
+			" must be <group_id>/<host_id> but got %s", d.Id())
```

Comment:
```suggestion
			" must be <group_id>/<id> but got %s", d.Id())
```
and check other sides

Reply:
OK